src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java

```diff
@@ -87,14 +88,25 @@ public String getSecretKeyVariable() {
         return secretKeyVariable;
     }
 
+    @DataBoundSetter
```

Comment:
You also need a matching public getter; and an update to `config.jelly`.

Reply:
Cheers. Just for my benefit, could you please point me to the docs explaining the need for both the setter and the getter together? Or just briefly explain why? I believe that it's necessary, but would appreciate a better understanding of it.

Reply:
[Docs](https://jenkins.io/doc/developer/forms/) are not really written I am afraid. The getter is needed so that the equivalent of `${instance.stsTokenDuration}` would work when reconfiguring an object from Jelly. Something like this would be evaluated by having a databound control inside an `f:entry`. You can get a test to fail against your code as written by using `JenkinsRule.configRoundtrip` with a `SecretBuildWrapper` (or `StepConfigTester` with a `BindingStep`, a bit more work), though it would be overkill if you are not doing anything out of the ordinary.

Reply:
Ah, okay. That makes sense. Thanks for the explanation :)